src/main/src/main/java/org/geoserver/template/FeatureWrapper.java

```diff
@@ -116,16 +116,10 @@ public FeatureWrapper(TemplateFeatureCollectionFactory templateFeatureCollection
     }
 
     private Catalog getCatalog() {
-        if (gsCatalog != null) return gsCatalog;
```

Comment:
Bizarre indeed... dunno, it's stuff that happened 11 and 8 years ago respectively, I'm guessing once upon a time there use to be a catalog2: https://github.com/geoserver/geoserver-history/blame/master/src/main/src/main/java/org/geoserver/template/FeatureWrapper.java#L129